stateMachine.js

```diff
@@ -114,18 +116,19 @@
           return curr && curr.name;
         }
 
-      , setState: function (name) {
+      , setState: function (name, args) {
           var prev = me.currentState(true);
           currentState = me._states[name];
 
-          me.onChange(me.currentState(), prev.name);
-          is(currentState.enter, 'Function') && currentState.enter();
+          me.onChange(me.currentState(), prev.name, args);
+          args.push(me.currentState());
```

Comment:
Why put the current state at the end of the list of arguments? Wouldn't it be more predictable to have the current state as the first argument?